lib/liquid.lua

```diff
@@ -2547,28 +2549,39 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get, error_handler )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
+    instance.error_handler = error_handler
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:genertic_get( location )
+    local error_handler = self.error_handler
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+        if not ret then
```

Comment:
yes, I got your point. But I think it is better let the user do the conversion and make it clear we expect string type as return result type. Because`  return tostring(ret)` will sometimes confuse users.
for example,
```
function user_get(location)
     -- do something 
    local result = {}. -- user mistakenly  return non-string type
    return result 
end

return tostring(ret). -- will get some string like this "table: 0x7f96bd700530", which is not a user want 
```